File: go/cpu/cpu_benchmark.go
// cpu_benchmark.go
package main

import (
	"fmt"
	"math"
	"time"
)

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	sqrtN := int(math.Sqrt(float64(n)))
	for i := 2; i <= sqrtN; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	start := time.Now()
	count := 0
	N := 1_000_000
	for i := 0; i < N; i++ {
		if isPrime(i) {
			count++
		}
	}
	fmt.Printf("Found %d primes in %dms\n", count, time.Since(start).Milliseconds())
}

File: node/cpu-benchmark.ts
// cpu-benchmark.js
const isPrime = (n: number): boolean => {
  if (n < 2) return false;
  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (n % i === 0) return false;
  }
  return true;
};

const start = Date.now();
let count = 0;
const N = 1_000_000;
for (let i = 0; i < N; i++) {
  if (isPrime(i)) count++;
}
console.log(`Found ${count} primes in ${Date.now() - start}ms`);
